pkgs/top-level/release-haskell.nix

```diff
@@ -275,6 +278,8 @@ let
           jobs.haskellPackages.hopenssl
           jobs.haskellPackages.hsemail
           jobs.haskellPackages.hsyslog
+          # simple regression test for agda which is updated via haskellPackages
+          jobs.nixosTests.agda
```

Comment:
I'm against adding this to `mergeable` unless there is someone on @NixOS/haskell that is willing to fix this when it breaks.  @sternenseemann would you be willing to do this?  If so, could you add yourself as a maintainer to `nixos/tests/agda.nix`?

----------------------------------

Other than that, this PR LGTM.

Reply:
I'm willing to have a look when it breaks.

Reply:
@turion Line 94 and 95 of this PR look like:

```nix
nixosTests.agda = (packagePlatforms pkgs.nixosTests).agda;
agdaPackages = packagePlatforms pkgs.agdaPackages;
```

This means that `nixosTests.agda` and `agdaPackages` will show up on the hydra build report, and the maintainers for these two derivations will get pinged (similar to how cedile shows up in https://github.com/NixOS/nixpkgs/pull/129606#issuecomment-879521411).  The maintainers will always be given at least 7 days to get these derivations working before `haskell-updates` is merged into `master`.

However, putting `agdaPackages` and `nixosTests.agda` into the `mergeable` job implies that these two derivations are "critical" and we can't merge `haskell-updates` without them passing.  This would mean that the @NixOS/haskell team is ultimately responsible with making sure they are working. (The `HACKING.md` has a little more info about the `mergeable` job: https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md.)

While I mostly agree that `nixosTests.agda` and `agdaPackages` are fine to go in our jobset, I don't agree that they should be added to `mergeable` for the following reasons:

- My guess is that the agda-related packages are not used by a large number of Nix users (unlike `hledger`, `niv`, `xmonad`, etc).  Unlike something like `pandoc`, I imagine there are not a lot of reverse dependencies of Agda in Nixpkgs.
- While `haskellPackages.Agda` is basically just a Haskell pacakge, `nixosTests.agda` and `agdaPackages` appear to sometimes fail with agda-related build problems.  I don't think it is reasonable to expect the NixOS/haskell team to figure out these problems and fix them.
- In general, I think we should reduce the number of packages in `mergeable` and instead rely on individual maintainers to fix packages.  I think this is the most sustainable position in the long term (as opposed to the NixOS/haskell team taking on more responsibility over time).

Reply:
Ah, thanks for pointing out that distinction. I wasn't aware of it. Yes, you are absolutely right. I'd like to be pinged (and possibly @alexarice and @neosimsim as well?), but `haskell-updates` shouldn't be delayed by a broken Agda, since not much depends on it.

Reply:
Removed from mergeable. One caveat is that the agda packages won't show up in the `maintained` job (which only checks `haskellPackages`). `hydra-report.hs` however should ping the maintainers if I'm not mistaken.